docs/lab/02/02.md

```diff
@@ -48,17 +48,187 @@ It is the role of the *IO Bank0* to multiplex and connect the peripherals to the
+Besides the implementation of the embedded HAL for different MCUs (RP2040 included), embassy-rs provides
+several functions like timers, BLE and network communication.
+
+<div align="center">
+![Rust EMbedded Stack](images/rust_embedded_stack.svg)
+</div>
+
+The crates used by Embassy-rs and their mapping are shown in the table bellow.
+
+| Crate | Position |
+|-------|----------|
+| [`embassy-executor`](https://docs.embassy.dev/embassy-executor/git/cortex-m/index.html) | Framework |
+| [`smoltcp`](https://docs.rs/smoltcp/latest/smoltcp/), [`defmt`](https://docs.rs/defmt/latest/defmt/) | Libraries |
+| [`embassy-net`](https://docs.embassy.dev/embassy-net/git/default/index.html), [`embassy-time`](https://docs.embassy.dev/embassy-time/git/default/index.html), [`embassy-usb`](https://docs.embassy.dev/embassy-usb/git/default/index.html), [`embassy-usb-logger`](https://docs.embassy.dev/embassy-usb-logger/git/default/index.html) | Framework Driver |
+| [`embassy-usb-driver`](https://docs.embassy.dev/embassy-usb-driver/git/default/index.html), [`embassy-time-driver`](https://docs.embassy.dev/embassy-time-driver/git/default/index.html) | Embassy HAL (API) |
+| [`cyw43`](https://docs.embassy.dev/cyw43/git/default/index.html), [`cyw43-pio`](https://docs.embassy.dev/cyw43-pio/git/default/index.html) | Driver (WiFi) |
+| [`embedded-hal`](https://docs.rs/embedded-hal/latest/embedded_hal/), [`embedded-hal-async`](https://docs.rs/embedded-hal-async/latest/embedded_hal_async/)| **Rust Embedded HAL (Standard)** |
+| [`embassy_rp`](https://docs.embassy.dev/embassy-rp/git/rp2040/index.html) | HAL Implementation |
+| [`cortex-m`](https://docs.rs/cortex-m/latest/cortex_m/), [`cortex-m-rt`](https://docs.rs/cortex-m-rt/latest/cortex_m_rt/) | μ-architecture crates |
+| [`rp_pac`](https://docs.embassy.dev/rp-pac/git/default/index.html) | Platform Access Crate |
+
+:::info
+
+The name *Embassy-rs* is derived form **Emb**edded **Asy**nchronous Rust.
+
+:::
+
+### Entry
+
+Embassy-rs is a framework built on top of `cortex-m-rt` and provides its own method of defining
+the entrypoint and bootloader.
+
+```rust
+use embassy_executor::Spawner;
+
+#[embassy_executor::main]
+async fn main(_spawner: Spawner) {
+    let peripherals = embassy_rp::init(Default::default());
+}
+```
+
+Instead of using the `entry` macro, embassy-rs provides the `embassy_executor::main` macro that
+sets the `main` function as the startup point and defines the correct bootloader.
+
+The `embassy_rp::init` function takes care of the peripheral initialization so that developers can jump
+right in and use them.
+
+:::note
+
+Embassy-rs is designed to work in an asynchronous way and this is why the `main` function is defined as `async`. For the time being, just take it as a must and ignore it.
+
+:::
+
+:::info
+
+The `main` function does not have to loop indefinitely, it is allowed to return, as it runs within the embassy-rs framework.
```

Comment:
This should be moved to the bare metal and explain that it has to loop.

Reply:
Done